src/core/sign.ml

```diff
@@ -15,17 +15,22 @@ type inductive =
   { ind_cons  : sym list  (** List of constructors                 *)
   ; ind_prop  : sym       (** Induction principle on propositions. *) }
 
+(** Notation properties of symbols. They are linked to symbols to provide
+    syntax extensions to these symbols. *)
+type notation =
+  | Prefix of unop (** Prefix (or unary) operator. *)
+  | Infix of binop (** Infix (or binary) operator. *)
+  | Quant (** Quantifier. *)
+
 (** Representation of a signature. It roughly corresponds to a set of symbols,
     defined in a single module (or file). *)
 type t =
   { sign_symbols  : (sym * Pos.popt) StrMap.t ref
   ; sign_path     : Path.t
   ; sign_deps     : (string * rule) list PathMap.t ref
   ; sign_builtins : sym StrMap.t ref
-  ; sign_unops    : (sym * unop ) StrMap.t ref
-  ; sign_binops   : (sym * binop) StrMap.t ref
-  ; sign_idents   : StrSet.t ref
-  ; sign_quants   : SymSet.t ref
+  ; sign_syntax   : notation SymMap.t ref
```

Comment:
I propose to use a pp_hint SymMap.t instead of a notation SymMap.t. It will ease interaction with sig_state. Then we can move the definition of pp_hint in sign.ml.

Reply:
Would you agree with renaming `pp_hint` into `notation`? I feel it as less technical, and more communicative in some way.